500 Error on viewing model

I had a model which the actions suggested I convert to a 3mf for performance reasons. When I try and access the model `/models/4fzh98dg46dl` it 500s with the following logs
Interestingly, I can view the `/models/4fzh98dg46dl/model_files/{id}` pages okay, and can access the data via the API without any 500 errors

```
{"method":"GET","path":"/models/4fzh98dg46dl","format":"html","controller":"ModelsController","action":"show","status":500,"allocations":122096,"duration":119.62,"view":0.0,"db":2.43,"@timestamp":"2026-06-28T11:12:56.904Z","@version":"1","message":"[500] GET /models/4fzh98dg46dl (ModelsController#show)"}
  
ActionView::Template::Error (undefined method 'upcase' for nil)
Caused by: NoMethodError (undefined method 'upcase' for nil)

Information for: ActionView::Template::Error (undefined method 'upcase' for nil):
    13:           <code><%= sanitize file.filename %></code>
    14:           <%- alts = file.alternative_formats %>
    15:           <% if alts.any? %> +
    16:             <%= safe_join(alts.map(&:extension).map { content_tag(:code) { it.upcase } }, ", ") %>
    17:           <% end %>
    18:         </small>
    19:       </div>
  
app/views/models/_file.html.erb:16
app/views/models/_file.html.erb:16
app/views/models/_file.html.erb:16:in 'Array#map'
app/views/models/_file.html.erb:16
app/views/models/show.html.erb:46
app/views/models/show.html.erb:43:in 'Hash#each_pair'
app/views/models/show.html.erb:43
app/controllers/application_controller.rb:141:in 'ApplicationController#switch_locale'

Information for cause: NoMethodError (undefined method 'upcase' for nil):
  
app/views/models/_file.html.erb:16
app/views/models/_file.html.erb:16
app/views/models/_file.html.erb:16:in 'Array#map'
app/views/models/_file.html.erb:16
app/views/models/show.html.erb:46
app/views/models/show.html.erb:43:in 'Hash#each_pair'
app/views/models/show.html.erb:43
app/controllers/application_controller.rb:141:in 'ApplicationController#switch_locale'
```

--- a/app/models/model.rb
+++ b/app/models/model.rb
@@ -272,7 +272,7 @@ class Model < ApplicationRecord
   end
 
   def file_extensions
-    model_files.map(&:extension).uniq
+    model_files.map(&:extension).uniq.compact
   end
 
   def size_on_disk

--- a/app/views/models/_file.html.erb
+++ b/app/views/models/_file.html.erb
@@ -13,7 +13,7 @@
           <code><%= sanitize file.filename %></code>
           <%- alts = file.alternative_formats %>
           <% if alts.any? %> +
-            <%= safe_join(alts.map(&:extension).map { content_tag(:code) { it.upcase } }, ", ") %>
+            <%= safe_join(alts.filter_map(&:extension).uniq.map { content_tag(:code) { it.upcase } }, ", ") %>
           <% end %>
         </small>
       </div>

